### PR TITLE
Microsoft.ML.Internal.Internallearn namespace requires certain internalization.

### LINKED ISSUE #2590: Microsoft.ML.Internal.Internallearn namespace requires certain internalization
At least
`public sealed class SlotDropper` should become internal.
`public static class PredictionUtil` should become internal
`public static class TypeUtils`
also all public delegates in this namespace can be made internal.

### PR DESCRIPTION
**Pull request by codemzs**
fixes #2590

### CODE REVIEW

**Comment by TomFinley** [comment]
>     public sealed class ReverseComparer<T> : IComparer<T>

How about this guy?

---
Refers to: src/Microsoft.ML.Data/Dirty/PredictionUtils.cs:160 in 1fe6806. [](commit_id = 1fe68062a524796b8438574b121af4c5e3463bf4, deletion_comment = False)

**Comment by TomFinley** [comment]
>     public sealed class ReverseComparer<T> : IComparer<T>

I also like the directory name `Dirty` here. :D

---
In reply to: [465734701](https://github.com/dotnet/machinelearning/pull/2667#issuecomment-465734701) [](ancestors = 465734701)

---
Refers to: src/Microsoft.ML.Data/Dirty/PredictionUtils.cs:160 in 1fe6806. [](commit_id = 1fe68062a524796b8438574b121af4c5e3463bf4, deletion_comment = False)

File: src/Microsoft.ML.Data/Dirty/PredictorInterfaces.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ namespace Microsoft.ML.Internal.Internallearn
     /// <summary>
     /// Signature for loading from a file name.
     /// </summary>
-    public delegate void SignaturePredictorFromFile(string fileName);
+    internal delegate void SignaturePredictorFromFile(string fileName);
```
Comment (yaeldekel) [inline src/Microsoft.ML.Data/Dirty/PredictorInterfaces.cs]: >SignaturePredictorFromFile [](start = 27, length = 26)

Is this used anywhere?

**Comment by Ivanidzo4ka** [comment]
>     public delegate void SignatureFeatureScorerTrainer();

touch me

---
Refers to: src/Microsoft.ML.Data/Dirty/PredictorInterfaces.cs:149 in a7a3170. [](commit_id = a7a317081c8e33865eadce0b3c6dbd43adc0f1d8, deletion_comment = False)

**Pull request merged**